core/src/main/java/com/linecorp/armeria/server/misc/InetAddressPredicates.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.misc;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableMap.Builder;
+
+/**
+ * A utility class which provides factory methods in order to easily create a {@link Predicate} of an
+ * {@link InetAddress}.
+ */
+public final class InetAddressPredicates {
+
+    /**
+     * A {@link Pattern} of a valid IP address.
+     */
+    private static final Pattern validIpV4Address =
+            Pattern.compile("^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}" +
+                            "(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$");
+
+    /**
+     * A {@link Splitter} for splitting an IP address.
+     */
+    private static final Splitter dotSplitter = Splitter.on('.').limit(4);
+
+    /**
+     * A table for converting an IP decimal value to the number of bits.
+     */
+    private static final Map<Integer, Integer> ipDecimalToMaskBits = new Builder<Integer, Integer>()
+            .put(0xFF, 8)
+            .put(0xFE, 7)
+            .put(0xFC, 6)
+            .put(0xF8, 5)
+            .put(0xF0, 4)
+            .put(0xE0, 3)
+            .put(0xC0, 2)
+            .put(0x80, 1)
+            .put(0x00, 0)
+            .build();
+
+    /**
+     * Returns a {@link Predicate} which returns {@code true} if the given {@link InetAddress} equals to
+     * the specified {@code address}.
+     *
+     * @param address the expected {@link InetAddress}
+     */
+    public static Predicate<InetAddress> ofExact(InetAddress address) {
+        requireNonNull(address, "address");
+        return address::equals;
```

Comment:
I guess we at least should perform some conversion between IPv4/IPv6 addresses? e.g.

- What if a user called this method with an IPv6 address which actually contains an IPv4 address?
- What if a user called this method with an IPv4 address and he or she called the returned predicate with an IPv6 address which actually contains an IPv4 address?

Perhaps you might want to reused some of the code I put in `DnsAddressEndpointGroup`: https://github.com/line/armeria/blob/b971e19d763cab0ace19a53cf0ea575576be58c4/core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsAddressEndpointGroup.java#L162-L195